client-server-server/src/main/scala/zio/raft/server/RaftServer.scala

```diff
@@ -0,0 +1,473 @@
+        connections = connections.updated(sessionId, SessionConnection(Some(routingId), expiresAt)),
+        routingToSession = routingToSession.updated(routingId, sessionId)
+      )
+    }
+    
+    def reconnect(sessionId: SessionId, routingId: RoutingId, now: Instant): Sessions = {
+      connections.get(sessionId) match {
+        case Some(conn) =>
+          val expiresAt = now.plusSeconds(90) // TODO: use config
+          // Remove old routing mapping if it exists
+          val cleanedRouting = conn.routingId.map(old => routingToSession.removed(old)).getOrElse(routingToSession)
+          copy(
+            connections = connections.updated(sessionId, conn.copy(routingId = Some(routingId), expiresAt = expiresAt)),
+            routingToSession = cleanedRouting.updated(routingId, sessionId)
+          )
+        case None => this
+      }
+    }
+    
+    def disconnect(sessionId: SessionId, routingId: RoutingId): Sessions =
+      copy(
+        connections = connections.updatedWith(sessionId)(_.map(_.copy(routingId = None))),
+        routingToSession = routingToSession.removed(routingId)
+      )
+    
+    def updateExpiry(sessionId: SessionId, now: Instant): Sessions = {
+      connections.get(sessionId) match {
+        case Some(conn) =>
+          val expiresAt = now.plusSeconds(90) // TODO: use config
+          copy(connections = connections.updated(sessionId, conn.copy(expiresAt = expiresAt)))
+        case None => this
+      }
+    }
+    
+    def removeExpired(now: Instant): (List[SessionId], Sessions) = {
+      val expired = connections.filter { case (_, conn) => conn.expiresAt.isBefore(now) }.keys.toList
+      val newMetadata = metadata -- expired
+      val newConnections = connections -- expired
+      val newRouting = routingToSession.filter { case (_, sid) => !expired.contains(sid) }
+      (expired, Sessions(newMetadata, newConnections, newRouting))
+    }
+    
+    def getRoutingId(sessionId: SessionId): Option[RoutingId] =
+      connections.get(sessionId).flatMap(_.routingId)
+    
+    def getMetadata(sessionId: SessionId): Option[SessionMetadata] =
+      metadata.get(sessionId)
+    
+    def findSessionByRouting(routingId: RoutingId): Option[SessionId] =
+      routingToSession.get(routingId)
+    
+    def closeAll(transport: ZmqServerTransport, leaderId: Option[MemberId]): UIO[Unit] = {
+      val reason = if (leaderId.isDefined) SessionCloseReason.NotLeaderAnymore else SessionCloseReason.Shutdown
+      ZIO.foreachDiscard(connections.values.flatMap(_.routingId)) { routingId =>
+        transport.sendMessage(routingId, SessionClosed(reason, leaderId)).orDie
+      }
+    }
+  }
+  
+  object Sessions {
+    def empty: Sessions = Sessions(Map.empty, Map.empty, Map.empty)
+    
+    def fromMetadata(metadata: Map[SessionId, SessionMetadata], config: ServerConfig): Sessions = {
+      val now = java.time.Instant.now() // Safe here as it's initialization
```

Comment:
java.time.Instant.now() is not allowed... you should accept now as parameter to the function